pythonforandroid/bootstraps/common/build/src/main/java/org/kivy/android/PythonService.java

```diff
@@ -116,7 +122,31 @@ protected void doStartForeground(Bundle extras) {
             builder.setSmallIcon(context.getApplicationInfo().icon);
             notification = builder.build();
         }
-        startForeground(1, notification);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O){
+            Log.v("python service","android 8: start foregrund with new channel");
+            startForegroundWithNewChannel(1, notification);
+        }else{
+            Log.v("python service","android <8: start foregrund withut new channel");
+            startForeground(1, notification);
+        }
+    }
+
+    /**
+     * https://stackoverflow.com/questions/47531742/startforeground-fail-after-upgrade-to-android-8-1
+     *
+     * necessary because from Android 8 on we need our own channel
+     */
+    private void startForegroundWithNewChannel(int id, Notification notification){
+        String NOTIFICATION_CHANNEL_ID = "org.bd.pureservice"; //TODO: dynamically
+        String channelName = "Pureservice";
+        NotificationChannel chan = new NotificationChannel(NOTIFICATION_CHANNEL_ID, channelName, NotificationManager.IMPORTANCE_NONE);
+        chan.setLightColor(Color.BLUE);
+        chan.setLockscreenVisibility(Notification.VISIBILITY_PRIVATE);
+        NotificationManager manager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+        assert manager != null;
```

Comment:
Minor question: why would we `assert` this rather than doing something else?
https://developer.android.com/reference/android/content/Context.html#getSystemService(java.lang.String)
> The service name or null if the class is not a supported system service.

So just wondering if we couldn't do something better about it. But the overall change is a great improvement anyway, so let's not bother much more if we can't think about something clever.